src/bucketizers.ts

```diff
@@ -278,62 +332,34 @@ export async function bucketize(
         const quads = new Parser().parse(x);
 
         const records = await extractor.parse_records(quads);
-        const relatedBuckets = new Map<string, Set<Term>>();
-        const requestedBuckets = new Map<string, Set<Term>>();
-        const includedBuckets = new Map<string, Set<Term>>();
+        const requestedBuckets = new Set<string>();
         const newMembers = new Map<string, Set<string>>();
 
+        const newRelations: {
+            origin: Bucket;
+            relation: BucketRelation;
+        }[] = [];
+
         for (const record of records) {
             const record_buckets = orchestrator.bucketize(
                 record,
                 buckets,
                 requestedBuckets,
                 newMembers,
+                newRelations,
                 sourceStream?.value || "root",
             );
+
+            record_buckets.forEach((x) => requestedBuckets.add(x));
```

Comment:
This record buckets should already be part of requested buckets by default, right? As you retrieve them in the bucketizers by using the getBucket function.

Reply:
Not necessarily, because record_buckets are strings :cry: